engine/src/flutter/shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -156,6 +160,9 @@ FlutterWindowsEngine::FlutterWindowsEngine(
     windows_proc_table_ = std::make_shared<WindowsProcTable>();
   }
 
+  engine_handle_ = next_engine_handle_++;
+  handle_to_engine_[engine_handle_] = this;
```

Comment:
My understanding is Windows lets you use different threads for different HWNDs. Should we have some sort of lock here?

Reply:
You can use HWND on the thread that it was created on (the thread must it's own message loop). All our HWNDs are created on main thread, and all embedder API assume main thread (same as other platforms). There's no need for locking. But we could maybe add thread affinity checks in future, at least for debug builds.

Reply:
> All our HWNDs are created on main thread, and all embedder API assume main thread (same as other platforms).

I don't think that's true. The developer is free to edit their app entrypoint to launch their window and Flutter Windows from some background thread. And the embedder API doesn't require that the platform thread is the main thread either, the embedder API just defers to the embedder's platform task runner. The Flutter Windows platform task runner should work on background threads too.

Reply:
I think I see what you mean. Technically there is no "main thread" on Windows. Main thread simply refers to the thread on which the engine is created (and to which the task runner HWND belongs). In theory it is possible to create two engines, each one of different thread and then pump message loop on both threads. That should work.

In any case the map is gone from the PR, so there is no need for lock. But you are right, we shouldn't assume that platform thread is same thread for all engine instances on windows.